qttypes/src/lib.rs

```diff
@@ -666,6 +749,265 @@ impl QVariant {
         })
     }
 
+    /// Wrapper around [`toByteArray()`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qvariant.html#toByteArray
+    pub fn to_qbytearray(&self) -> QByteArray {
+        cpp!(unsafe [self as "const QVariant*"] -> QByteArray as "QByteArray" {
+            return self->toByteArray();
+        })
+    }
+
+    /// Returns the variant as a Rust's native [`Vec`]`<`[`u8`]`>` if the variant is supported by
+    /// [`to_qbytearray()`] method, otherwise returns an empty vector.
+    ///
+    /// Unlike [`to_qbytearray()`] which might use [implicit sharing] (copy-on-write), this method
+    /// always allocates memory for a [`Vec`].
+    ///
+    /// [`to_qbytearray()`]: Self::to_qbytearray()
+    /// [implicit sharing]: https://doc.qt.io/qt-5/implicit-sharing.html
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let bytearray = self.to_qbytearray();
+        bytearray.to_slice().to_vec()
+    }
+
+    // TODO: toChar(), requires QChar wrapper
+
+    /// Wrapper around [`toDate()`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qvariant.html#toDate
+    pub fn to_date(&self) -> QDate {
+        cpp!(unsafe [self as "const QVariant*"] -> QDate as "QDate" {
+            return self->toDate();
+        })
+    }
+
+    /// Wrapper around [`toDateTime()`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qvariant.html#toDateTime
+    pub fn to_date_time(&self) -> QDateTime {
+        cpp!(unsafe [self as "const QVariant*"] -> QDateTime as "QDateTime" {
+            return self->toDateTime();
+        })
+    }
+
+    /// Wrapper around [`toDouble(bool *ok)`][method] method.
+    ///
+    /// [method]: https://doc.qt.io/qt-5/qvariant.html#toDouble
+    pub fn to_double(&self) -> Option<f64> {
```

Comment:
I guess i'd call this `to_f64` instread

Reply:
then, it would apply to all the others in family as well